test/AutoDiff/superset_adjoint.swift

```diff
@@ -35,9 +35,15 @@ SupersetAdjointTests.test("CrossModuleClosure") {
   expectEqual(1, gradient(at: Float(1)) { x in x + 2 })
 }
 
-SupersetAdjointTests.test("CrossModule") {
-  expectEqual((1, 1), gradient(at: 1, 2, in: (+) as (Float, Float) -> Float))
-}
+// FIXME: The expression `(+) as @autodiff (Float, @nondiff Float) -> Float)`
+// forms a curry thunk of `Float.+` before conversion to @autodiff, and AD
+// doesn't know how to differentiate the curry thunk, so it produces a
+// "function is not differentiable" error.
+// FIXME: Propagate wrt indices correctly so that this actually takes the
+// gradient wrt only the first parameter, as intended.
+// SupersetAdjointTests.test("CrossModule") {
+//   expectEqual(1, gradient(at: 1, 2, in: (+) as @autodiff (Float, @nondiff Float) -> Float))
```

Comment:
Actually, this will not work semantically. The arity of the differential operator determines the number of results returned by the pullback. We can't expect to exhaustively define differential operators for arbitrary combinations of `@nondiff` and non-`@nondiff` parameters.

Therefore, `gradient(at: 1, 2, in: ...)`, gives you a pullback that returns a tuple. To obtain VJPs only w.r.t. the first parameter, you need to pass the single argument in place of `at:` and pass non-diff arguments as constants, i.e. `gradient(at: 1, in: { x in x + 2 })`. The function being passed to `Builtin.autodiffApply` will not have a `@nondiff` on its parameter types.